File: julia/raindrops/raindrops.jl
function raindrops(number::Int)::String 
    drops = [ (3, "Pling"), (5, "Plang"), (7, "Plong")]
    sound_of_drops(number, drop, drop_sound) = rem(number, drop) == 0 ? drop_sound : ""
    default_sound(number, drop_sound) = drop_sound  == "" ? string(number) : drop_sound

    (drops
      |> ds->map(drop->sound_of_drops(number, drop...), ds) 
      |> join 
      |> drop_sound->default_sound(number, drop_sound))
end

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let drops =
    [ (3, "Pling")
      (5, "Plang")
      (7, "Plong") ]

let is_divisible_by dividend divisor = (dividend % divisor) = 0

let sound_of_drops number (drop, drop_sound) =
    match is_divisible_by number drop with
    | true -> drop_sound
    | false -> ""

let default_sound number rain =
    match rain with
    | "" -> number.ToString()
    | rain -> rain

let convert (number: int) : string =
    drops
    |> List.map (sound_of_drops number)
    |> String.concat ""
    |> default_sound number
